ui/src/organizations/containers/OrgDashboardsIndex.tsx

```diff
@@ -33,73 +34,110 @@ interface RouterProps {
 
 interface DispatchProps {
   notify: NotificationsActions.PublishNotificationActionCreator
+  getDashboards: typeof getDashboardsAction
+  populateDashboards: typeof populateDashboardsAction
 }
 
 interface StateProps {
   org: Organization
+  dashboards: Dashboard[]
 }
 
 type Props = WithRouterProps & RouterProps & DispatchProps & StateProps
 
+interface State {
+  loadingState: RemoteDataState
+}
+
 @ErrorHandling
-class OrgDashboardsIndex extends Component<Props> {
+class OrgDashboardsIndex extends Component<Props, State> {
+  public state = {
+    loadingState: RemoteDataState.NotStarted,
+  }
+
+  public componentDidMount = async () => {
+    this.setState({loadingState: RemoteDataState.Loading})
+
+    const {getDashboards, org} = this.props
+
+    await getDashboards(org.id)
```

Comment:
What do you think about putting all this fetching logic in a separate data-fetching component?

Reply:
I will look in to this idea, but won't do it in this PR